src/main/java/org/mule/extension/ftp/internal/source/FtpDirectoryListener.java

```diff
@@ -248,22 +243,22 @@ private FtpFileSystem openConnection() throws Exception {
     return ftpFileSystemConnection;
   }
 
-  private boolean processFile(Result<InputStream, FtpFileAttributes> file,
-                              PollContext<InputStream, FtpFileAttributes> pollContext) {
+  private boolean processFile(Result<String, FtpFileAttributes> file,
```

Comment:
The path from the list operation can be turn into the InputStream inside this method, or it can be read before and sent into this method already as an InputStream as it was before.

Reply:
yes added the read call before and passing InputStream as it was before.